crates/red_knot_workspace/src/site_packages.rs

```diff
@@ -100,55 +244,341 @@ fn site_packages_dir_from_sys_prefix(
+                    let line_number = index.checked_add(1).and_then(NonZeroUsize::new).unwrap();
+                    return Err(SitePackagesDiscoveryError::PyvenvCfgParseError(
+                        pyvenv_cfg_path,
+                        PyvenvCfgParseErrorKind::TooManyEquals { line_number },
+                    ));
+                };
+                let key = key.trim();
+                let value = value.trim();
+                match key {
+                    "include-system-site-packages" => {
+                        include_system_site_packages = value.eq_ignore_ascii_case("true");
+                    }
+                    "home" => base_executable_home_path = Some(value),
+                    // `virtualenv` and `uv` call this key `version_info`,
+                    // but the stdlib venv module calls it `version`
+                    "version" | "version_info" => version_info_string = Some(value),
+                    _ => continue,
+                }
+            }
+        }
+
+        // The `home` key is read by the standard library's `site.py` module,
+        // so if it's missing from the `pyvenv.cfg` file
+        // (or the provided value is invalid),
+        // it's reasonable to consider the virtual environment irredeemably broken.
+        let Some(base_executable_home_path) = base_executable_home_path else {
+            return Err(SitePackagesDiscoveryError::PyvenvCfgParseError(
+                pyvenv_cfg_path,
+                PyvenvCfgParseErrorKind::NoHomeKey,
+            ));
+        };
+        let Some(base_executable_home_path) =
+            PythonHomePath::new(base_executable_home_path, system)
+        else {
+            return Err(SitePackagesDiscoveryError::PyvenvCfgParseError(
+                pyvenv_cfg_path,
+                PyvenvCfgParseErrorKind::HomeValueIsNotADirectory,
+            ));
+        };
+
+        // but the `version`/`version_info` key is not read by the standard library,
+        // and is provided under different keys depending on which virtual-environment creation tool
+        // created the `pyvenv.cfg` file. Lenient parsing is appropriate here:
+        // the file isn't really *invalid* if it doesn't have this key,
+        // or if the value doesn't parse according to our expectations.
+        let minor_version = version_info_string.and_then(|version_string| {
+            let mut version_info_parts = version_string.split('.');
+            if version_info_parts.next()? != "3" {
+                return None;
+            }
+            let minor_version_string = version_info_parts.next()?;
+            PythonMinorVersion::from_version_string(minor_version_string)
+        });
+
+        let metadata = Self {
+            venv_path,
+            base_executable_home_path,
+            include_system_site_packages,
+            minor_version,
+        };
+
+        tracing::trace!("Resolved metadata for virtual environment: {metadata:?}");
```

Comment:
Considering that we log `Attempting to...` with level=`debug`, I would expect that we log the "outcome" at the same level. To me, knowing to what the virtual environment resolved to seems more important than knowing that Red Knot is about to resolve it. 

However, That would mean that we might not want log the entire `metadata` struct in debug mode and have to write a slightly more polished message.

Reply:
> Considering that we log `Attempting to...` with level=`debug`, I would expect that we log the "outcome" at the same level. To me, knowing to what the virtual environment resolved to seems more important than knowing that Red Knot is about to resolve it.

We do log that with `level=debug`, but we emit one log message once we've collected all `site-packages` directories, at the end of `site_packages_dirs()` (which I'm about to rename to `site_packages_directories()`. This makes for a less busy debug log output that's easier to understand: just one log message that tells you about both `site-packages` entries discovered, instead of two log messages that might have other logs in between them.

Reply:
Sounds good? Just make sure to not log a struct with multiple fields with level `debug`. The output is intended for users